src/02-crm/companies.apib

```diff
@@ -112,6 +112,12 @@ Get details for a single company.
             + added_at: `2016-02-04T16:44:33+00:00` (string)
             + updated_at: `2016-02-05T16:44:33+00:00` (string)
             + tags: prospect, expo (array[string])
+            + custom_fields (array)
+                + (object)
+                    + field: (object)
+                        + type: `customField` (string)
+                        + id: `bf6765de-56eb-40ec-ad14-9096c5dc5fe1` (string)
+                    + value: `092980616` (string)
```

Comment:
I'm also not sure about this. If it's a custom field that links to a contact (or other "other_id"), it should probably return the type and the id inside an object? // @jenssegers @stivni @mrtus

Reply:
Custom field layout should probably depend on the type of the custom field.

Reply:
but don't you always have to choose the type `customer_id` for example?

Reply:
But we don't pass the type of the custom field. I would expect to be able to see if I get a scalar value, a multiselect or an entity here, without having to query customFieldDefinition endpoints.

```
{
    "custom_fields": [
        {
            "field": {
                "type": "customFieldDefinition",
                "id": "bf6765de-56eb-40ec-ad14-9096c5dc5fe1"
            },
            "value": {
                "type": "string",
                "value": "092980616"
            }
        },
        {
            "field": {
                "type": "customFieldDefinition",
                "id": "bf6765de-56eb-40ec-ad14-9096c5dc5fe1"
            },
            "value": {
                "type": "multipleSelect",
                "value": "selected-value"
            }
        },
        {
            "field": {
                "type": "customFieldDefinition",
                "id": "bf6765de-56eb-40ec-ad14-9096c5dc5fe1"
            },
            "value": {
                "type": "contact",
                "value": "bf6765de-56eb-40ec-ad14-9096c5dc5fe1" // contact id
            }
        }
    ],
}
```

This might be a way to return all types in the same structure with the type information available?

Reply:
Doesn't look bad! But I am wondering if we really need the type (i'd say yes go for it, you can see it at a glance when using the api) but the value type is also already defined in the cf definition. To be discussed atleast 👍 But I do like this aproach! This would surely help integrators with parsing custom fields.

Reply:
The double "value" is a bit strange though. The type might be more logical in the "field" array 🤔